### ISSUE TITLE
Is this possible to be used with multiple open pose controlnet?`

### ISSUE THREAD

**Issue by strouder** [hako-mikan/sd-webui-regional-prompter#23]
I have controlnet 0 (with an openpose) + ControlNet 1 (another openpose)... 
I want to combine the two poses into 1 image divided in the center but the characters keep on merging into 1 mutation.

**Comment by Kanareika** [comment]
Quick dirty fix: put both your poses on one image and feed it into one openpose tab.
Also, there is problem with Style Adapter (https://github.com/Mikubill/sd-webui-controlnet/discussions/700). Looks like the extensions don't mix well.

**Comment by Symbiomatrix** [comment]
Are you by any chance using the standard BREAK syntax? I seem to be experiencing similar issues on that, but not with multi controlnet, on a simple test with 2D syntax and vertical mode. (See #19 for details.)
Need to look into that later.

![RegionalMcontrol](https://user-images.githubusercontent.com/41131377/229385525-076bf384-7b90-4f09-9228-e874c88917dd.png)

**Comment by Symbiomatrix** [comment]
I seem to be able to reproduce the phenomenon with the BREAK syntax, yet with 2D + vertical everything seems to be working quite well.
My guess is that controlnet might somehow be affected by the raw BREAK when it builds the mask (it seems to operate first, possibly due to directory names in the extensions call stack), whereas in 2D mode the conversion of the special keywords to BREAK only happens once regional's process is called later, thus controlnet is unaffected. But this needs closer inspection and I don't understand controlnet's workings or extension interactions nearly well enough. For now, 2D syntax is not a bad workaround, I think.

![RegionalMcontrol](https://user-images.githubusercontent.com/41131377/229386254-cd9f7640-2cca-4c39-b2b2-9130ae18e2c6.png)

**Comment by Kanareika** [comment]
@Symbiomatrix I'm sorry for vague explanation. When Regional Prompter is active, style adapter changes does not appy at all:
1. Without RP active:
![Снимок экрана 2023-04-03 074324](https://user-images.githubusercontent.com/10144329/229410028-d78e3342-d005-4f28-83a9-2f35d25ca9f7.png)
2. With RP active:
![Снимок экрана 2023-04-03 074246](https://user-images.githubusercontent.com/10144329/229410112-b8ef5552-d1c3-4f32-a5aa-edc7eed9ce6b.png)

Changing syntax make no difference (same result with BREAK and ADD). "Base Ratio" also does not affect. But, if using "Latent" generation mode, first half of a batch is kinda affected.

**Comment by Symbiomatrix** [comment]
@Kanareika Are you getting this message when the run starts with region + style?
"Warning: StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference"

**Comment by Kanareika** [comment]
@Symbiomatrix No. For me, this message appears only if i don't batch generations (you can see on my screenshots, that i use batch of 2) with Style Adapter. Regional Prompter does not trigger this message.

**Comment by Symbiomatrix** [comment]
> @Symbiomatrix No. For me, this message appears only if i don't batch generations (you can see on my screenshots, that i use batch of 2) with Style Adapter. Regional Prompter does not trigger this message.

Indeed, I have wondered about the batch; but that seems to be a sort of suppression, rather than a fix for the underlying issue;
For me, when running style control with batchsize 1 and no region, the warning doesn't appear - is this the same for you?

**Comment by Kanareika** [comment]
> is this the same for you?
Yes. 
I remember getting this message with bathsize 1, but it was probably due to exceeding the token limit of 75

**Comment by Symbiomatrix** [comment]
@Kanareika Alright, it's not (technically) a bug in regional but in controlnet - the BREAK keyword (to which all region separators are converted) is buggy in style mode, as is going over 75 tokens. You were there, so I guess you're better aware of the issue than I am.

https://github.com/Mikubill/sd-webui-controlnet/issues/502

**Comment by Symbiomatrix** [comment]
@Kanareika Huh, whaddya know. It was indeed almost the exact same fix as regional's pos/neg split.
I'm on an old version of controlnet, but I'll try to upgrade and see if it's still structured the same.

**Comment by Symbiomatrix** [comment]
Great success. This needs two PRs to work, one on hako's repo and the other on miku's. It's frankly a bit of a patchwork, but it seems to get the job done.

![BugRegionStyle](https://user-images.githubusercontent.com/41131377/230632948-bae0809f-1080-43b8-9d6c-c4a12843aa64.png)

**Comment by Symbiomatrix** [comment]
Here are the necessary PRs:
https://github.com/Mikubill/sd-webui-controlnet/pull/717
(Done) https://github.com/hako-mikan/sd-webui-regional-prompter/pull/40
